File: app/src/main/java/com/sample/app/fragment_with_expandableview/adapters/ExpandableAdapter.java
package com.sample.app.fragment_with_expandableview.adapters;

import android.content.Context;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseExpandableListAdapter;
import android.widget.TextView;

import com.sample.app.R;
import com.sample.app.entity.Customer;

import java.util.List;

public class ExpandableAdapter extends BaseExpandableListAdapter {
    private Context context;
    private List<Customer> itemList;

    public ExpandableAdapter(Context context, List<Customer> itemList) {
        this.context = context;
        this.itemList = itemList;
    }

    @Override
    public Object getChild(int groupPosition, int childPosition) {
        List<Customer> childList = itemList;
        return childList.get(groupPosition);
    }

    @Override
    public long getChildId(int groupPosition, int childPosition) {
        return childPosition;
    }

    @Override
    public View getChildView(int groupPosition, int childPosition, boolean isLastChild,
                             View view, ViewGroup parent) {

        Customer item = (Customer) getChild(groupPosition, childPosition);
        if (view == null) {
            LayoutInflater infalInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            view = infalInflater.inflate(R.layout.cust_customer_list, null);
        }

        TextView nameView = (TextView) view.findViewById(R.id.nameTextView);
        nameView.setText((item.getName()));
        TextView cityView = (TextView) view.findViewById(R.id.cityTextView);
        cityView.setText((item.getCity()));
        TextView countryView = (TextView) view.findViewById(R.id.countryTextView);
        countryView.setText((item.getCountry()));

        return view;
    }

    @Override
    public int getChildrenCount(int groupPosition) {
        return 1;

    }

    @Override
    public Object getGroup(int groupPosition) {
        return itemList.get(groupPosition);
    }

    @Override
    public int getGroupCount() {
        return itemList != null ? itemList.size() : 0;
    }

    @Override
    public long getGroupId(int groupPosition) {
        return groupPosition;
    }

    @Override
    public View getGroupView(int groupPosition, boolean isLastChild, View view,
                             ViewGroup parent) {

        Customer item = (Customer) getGroup(groupPosition);
        if (view == null) {
            LayoutInflater inf = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            view = inf.inflate(R.layout.expandable_heading, null);
        }

        TextView heading = (TextView) view.findViewById(R.id.heading);
        heading.setText(item.getName());

        return view;
    }

    @Override
    public boolean hasStableIds() {
        return true;
    }

    @Override
    public boolean isChildSelectable(int groupPosition, int childPosition) {
        return true;
    }

    public void setDataList(List<Customer> itemList){
        this.itemList = itemList;
    }

}

File: app/src/main/java/com/sample/app/fragment_with_expandableview/adapters/ExpandableAdapter.kt
package com.sample.app.fragment_with_expandableview.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseExpandableListAdapter
import android.widget.TextView

import com.sample.app.R
import com.sample.app.entity.Customer

class ExpandableAdapter(private val context: Context, private var itemList: List<Customer>?) : BaseExpandableListAdapter() {

    override fun getChild(groupPosition: Int, childPosition: Int): Any {
        val childList = itemList
        return childList!![groupPosition]
    }

    override fun getChildId(groupPosition: Int, childPosition: Int): Long {
        return childPosition.toLong()
    }

    override fun getChildView(groupPosition: Int, childPosition: Int, isLastChild: Boolean,
                              view: View?, parent: ViewGroup): View {
        var view = view

        val item = getChild(groupPosition, childPosition) as Customer
        if (view == null) {
            val infalInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            view = infalInflater.inflate(R.layout.item_customer, null)
        }

        val nameView = view!!.findViewById(R.id.nameTextView) as TextView
        nameView.text = item.name
        val cityView = view.findViewById(R.id.cityTextView) as TextView
        cityView.text = item.city
        val countryView = view.findViewById(R.id.countryTextView) as TextView
        countryView.text = item.country

        return view
    }

    override fun getChildrenCount(groupPosition: Int): Int {
        return 1

    }

    override fun getGroup(groupPosition: Int): Any {
        return itemList!![groupPosition]
    }

    override fun getGroupCount(): Int {
        return if (itemList != null) itemList!!.size else 0
    }

    override fun getGroupId(groupPosition: Int): Long {
        return groupPosition.toLong()
    }

    override fun getGroupView(groupPosition: Int, isLastChild: Boolean, view: View?,
                              parent: ViewGroup): View {
        var view = view

        val item = getGroup(groupPosition) as Customer
        if (view == null) {
            val inf = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            view = inf.inflate(R.layout.expandable_heading, null)
        }

        val heading = view!!.findViewById(R.id.heading) as TextView
        heading.text = item.name

        return view
    }

    override fun hasStableIds(): Boolean {
        return true
    }

    override fun isChildSelectable(groupPosition: Int, childPosition: Int): Boolean {
        return true
    }

    fun setDataList(itemList: List<Customer>) {
        this.itemList = itemList
    }

}
